crypto/bls12381/key_test.go

```diff
@@ -167,3 +170,44 @@ func TestPubKey_MarshalJSON(t *testing.T) {
 	err = pubKey2.UnmarshalJSON(jsonBytes)
 	require.NoError(t, err)
 }
+
+func TestPubKey_NewPublicKeyFromBytes(t *testing.T) {
+	unmarshalErr := errors.New("could not unmarshal bytes into pubkey")
+
+	unmarshal := func(s string) ([]byte, error) {
+		type blstPublicKey = blst.P1Affine
+
+		bz, err := hex.DecodeString(s)
+		if err != nil {
+			return nil, err
+		}
+		pk := new(blstPublicKey).Uncompress(bz)
+		if pk == nil {
+			return nil, unmarshalErr
+		}
+		pkc := pk.Serialize()
+		if pkc == nil {
+			return nil, errors.New("could not serialize pubkey")
+		}
+		return pkc, nil
+	}
+
+	invalidPubKeys := map[string]string{
+		"NotInG1":   "8123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef",
+		"InfFalseB": "800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
```

Comment:
This is a valid key, right? At least, the test passes with it even without the introduced changes.

Reply:
I copied it from the report. I think it means "infinite point A". The other one is "infinite point B". A and B are two parts. 

> One of the initially non-obvious things about BLS12-381 is that we're really dealing with two curves, not one.
https://hackmd.io/@benjaminion/bls12-381

Reply:
But shouldn't it fail or not?

Reply:
I still think it's helpful to keep them even though they don't yield an error on `main`.